src/platform_impl/macos/view.rs

```diff
@@ -504,30 +561,29 @@ extern "C" fn insert_text(this: &Object, _sel: Sel, string: id, _replacement_ran
         let state_ptr: *mut c_void = *this.get_ivar("winitState");
         let state = &mut *(state_ptr as *mut ViewState);
 
-        let has_attr: BOOL = msg_send![string, isKindOfClass: class!(NSAttributedString)];
-        let characters = if has_attr != NO {
-            // This is a *mut NSAttributedString
-            msg_send![string, string]
-        } else {
-            // This is already a *mut NSString
-            string
-        };
-
-        let slice =
-            slice::from_raw_parts(characters.UTF8String() as *const c_uchar, characters.len());
-        let string = str::from_utf8_unchecked(slice);
+        let string = id_to_string(string);
 
         // We don't need this now, but it's here if that changes.
         //let event: id = msg_send![NSApp(), currentEvent];
 
-        let mut events = VecDeque::with_capacity(characters.len());
+        if state.ime_state.is_some() {
+            AppState::queue_event(EventWrapper::StaticEvent(Event::WindowEvent {
+                window_id: WindowId(get_window_id(state.ns_window)),
+                event: WindowEvent::IME(IME::Commit(string.clone())),
+            }));
+            state.ime_state = None;
+            AppState::queue_event(EventWrapper::StaticEvent(Event::WindowEvent {
+                window_id: WindowId(get_window_id(state.ns_window)),
+                event: WindowEvent::IME(IME::Disabled),
```

Comment:
The idea behind `IME::Disabled` is when your IME stops completely, so if you had some text in flight you'll abort inserting and also stop providing anything to IME. For example if you have two layouts, one is `en(us)` and the other one is `mozc` input, switching from `en(us)` to `mozc` should trigger `IME::Enabled`, since you don't need IME for `en(us)`, and so switching from `mozc` to `en(us)` should issue `IME::Disabled`, but not when you're editing with IME.

If you don't know when user triggers IME you should likely send IME::Enabled at start and then one on exit of a particular window. Though, I'd assume that there's a way to determine whether there's IME or not on macOS.

Reply:
I surveyed how to handle the notification about what IME has been changed, and I found this would be very difficult.

In order to do this, you have to implement protocol `NSTextInputClient` and class `NSTextInputContext`. Here protocol in Objective-C is like trait in Rust. 

About `NSTextInputClient`, you have to implement some functions like `setMarkedText`, `insertText` and `validAttributesForMarkedText` with subclassing `NSView`. This is described in [Apple's official developer guide about NSTextInputClient](https://developer.apple.com/documentation/appkit/nstextinputclient#overview). So now you can handle marked text with already implemented functions.

But about `NSTextInputContext`, this is used to handle the information about what IME has been changed. This class is needed to catch notification about IME. This is still not implemented in [cocoa](https://github.com/servo/core-foundation-rs/tree/master/cocoa).

That's why I only remove `IME::Disabled` in this PR.